client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-upgrade-plan/test/index.tsx

```diff
@@ -1,34 +1,50 @@
 /**
  * @jest-environment jsdom
  */
-import { screen } from '@testing-library/react';
+import { PLAN_MIGRATION_TRIAL_MONTHLY } from '@automattic/calypso-products';
+import { screen, waitFor } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
+import nock from 'nock';
 import React from 'react';
 import useAddHostingTrialMutation from 'calypso/data/hosting/use-add-hosting-trial-mutation';
-import useCheckEligibilityMigrationTrialPlan from 'calypso/data/plans/use-check-eligibility-migration-trial-plan';
 import { useSite } from 'calypso/landing/stepper/hooks/use-site';
 import plansReducer from 'calypso/state/plans/reducer';
 import SiteMigrationUpgradePlan from '../';
 import { StepProps } from '../../../types';
 import { mockStepProps, renderStep } from '../../test/helpers';
 
+const planSlug = PLAN_MIGRATION_TRIAL_MONTHLY;
+
 jest.mock( 'calypso/landing/stepper/hooks/use-site' );
-jest.mock( 'calypso/data/plans/use-check-eligibility-migration-trial-plan' );
 jest.mock( 'calypso/data/hosting/use-add-hosting-trial-mutation' );
 
 ( useSite as jest.Mock ).mockReturnValue( {
 	ID: 'site-id',
 	URL: 'https://site-url.wordpress.com',
 } );
 
-( useCheckEligibilityMigrationTrialPlan as jest.Mock ).mockReturnValue( {
-	data: { eligible: true },
-} );
-
 ( useAddHostingTrialMutation as jest.Mock ).mockImplementation( ( { onSuccess } ) => ( {
 	addHostingTrial: () => onSuccess(),
 } ) );
 
+const mockApi = () => nock( 'https://public-api.wordpress.com:443' );
+
+const API_RESPONSE_EMAIL_VERIFIED = {
+	eligible: true,
+	error_code: 'email-verified',
+};
+
+const API_RESPONSE_EMAIL_UNVERIFIED = {
+	eligible: false,
+	error_code: 'email-unverified',
+};
+
+const mockTrialEligibilityAPI = ( payload: any ) => {
```

Comment:
Nit: is it worth typing the payload here? Even if it's just `Record<string, string | boolean >`?

I think it would be worth creating a type like the following:
```ts
type TrialEligibilityResponse {
	eligible: boolean;
	error_code: string; // we could also be stricter about the values here
}
```